File: togglebuttonlayout/src/main/java/com/savvyapps/togglebuttonlayout/Toggle.java
package com.savvyapps.togglebuttonlayout;

import android.graphics.drawable.Drawable;
import android.support.annotation.NonNull;

/**
 * An item within a {@link ToggleButtonLayout}
 */
public class Toggle {

    private int id;
    private Drawable icon;
    private CharSequence title;
    protected boolean selected;

    public Toggle(int id, @NonNull Drawable icon, @NonNull CharSequence title) {
        if (id == 0) {
            throw new IllegalArgumentException("Toggle must have a non-zero id");
        }
        this.id = id;
        this.icon = icon;
        this.title = title;
    }

    public int getId() {
        return id;
    }

    public Drawable getIcon() {
        return icon;
    }

    public CharSequence getTitle() {
        return title;
    }

    public boolean isSelected() {
        return selected;
    }
}

File: togglebuttonlayout/src/main/java/com/savvyapps/togglebuttonlayout/Toggle.kt
package com.savvyapps.togglebuttonlayout

import android.graphics.drawable.Drawable

/**
 * An item within a [ToggleButtonLayout]
 */
class Toggle(

        /**
         * The ID provided within the Toggle
         */
        val id: Int,

        /**
         * The icon drawable inflated by the menu resource or passed during creation
         */
        val icon: Drawable?,

        /**
         * Optional title
         */
        val title: CharSequence?) {

    /**
     * The selection state of the toggle
     */
    var isSelected: Boolean = false


    /**
     * Get the parent view of the toggle. Null if not yet inflated
     */
    var toggleButtonLayout: ToggleButtonLayout? = null
        internal set

    /**
     * The current toggle view. Null if not yet inflated
     */
    var toggleView: ToggleView? = null
        internal set

    init {
        if (id == 0) {
            throw IllegalArgumentException("Toggle must have a non-zero id")
        }
    }
}
